services/algolia-sync/src/components/algolia/helpers.js

```diff
@@ -0,0 +1,47 @@
+// Build out the sections and remove duplicates.
+// Example: toplevel > secondlevel > thirdlevel
+const buildSections = async (c) => {
+  // Get the values of the seconds
+  const sections = await c.websiteSchedules.map(s => (s.section.hierarchy.map(n => n.fullName)));
+  // flatten the array and remove duplicates.
+  const sectionsNames = [...new Set([].concat(...sections))];
+  // format and build out section structure
+  return sectionsNames.reduce((obj, name) => {
+    const number = [name.split('>').length - 1];
+    const key = `lev${number}`;
+    const values = obj[key] || [];
+    return { ...obj, [key]: [...values, name] };
+  }, {});
+};
+
+// Slightly boost companies and maybe others later
+const boostResult = async (node) => {
+  if (node.type === 'company') {
+    return 1;
+  }
+  return 0;
+};
+
+// Formats the object for Algolia's bulk update
+const buildObj = async (nodes, tenant) => Promise.all(nodes.map(async (node) => {
```

Comment:
The map doesn't need to be wrapped in `Promise.all` since the `buildSections` and `boostResult` functions should no longer be async functions, therefore they do not need to be awaited.